gem/optimise.py

```diff
@@ -35,13 +35,11 @@ def literal_rounding(node, self):
 
 @literal_rounding.register(Literal)
 def literal_rounding_literal(node, self):
-    if not node.shape:
-        return node  # skip scalars
     table = node.array
     epsilon = self.epsilon
     # Mimic the rounding applied at COFFEE formatting, which in turn
     # mimics FFC formatting.
-    one_decimal = numpy.round(table, 1)
+    one_decimal = numpy.asarray(numpy.round(table, 1))
```

Comment:
Apparently, `numpy` likes to turn zero-dimensional arrays into scalars, so we undo that here with this cast.